provisioning/provisioning.go

```diff
@@ -32,6 +33,29 @@ type SubmitConfig struct {
 	DeleteSession bool           // explicitly DELETE the session object after we are done
 }
 
+// SetClient sets the HTTP(s) client connection configuration
+func (cfg *SubmitConfig) SetClient(client *common.Client) error {
+	if client == nil {
+		return fmt.Errorf("no client supplied")
+	}
+	cfg.Client = client
+	return nil
+}
+
+// SetSubmitURI sets the URI Parameter
+func (cfg *SubmitConfig) SetSubmitURI(uri string) error {
+	u, err := url.Parse(uri)
+	if err != nil || !u.IsAbs() {
+		return fmt.Errorf("malformed URI")
```

Comment:
suggest to split the conditions "err != nil" and "malformed URI" and report separately

Reply:
Ok, makes sense!